cogs/statistics.py

```diff
@@ -12,6 +14,137 @@ def __init__(self, bot):
+    async def statistics(self, ctx):
+        """Statistic related commands"""
+        if ctx.invoked_subcommand is None:
+            await self.bot.send_cmd_help(ctx)
+
+    @statistics.command(name="user")
+    async def statistics_user(self, ctx):
+        """Statistics of the user"""
+        await ctx.trigger_typing()
+        cursor = self.db.commands.find({"author": ctx.author.id})
+        data = discord.Embed(
+            description="Command statistics of {0}".format(ctx.author))
+        data = await self.generate_embed(ctx, data, cursor, False)
+        try:
+            await ctx.send(embed=data)
+        except discord.Forbidden:
+            await ctx.send("Need permission to embed links")
+
+    @statistics.command(name="guild")
+    async def statistics_guild(self, ctx):
+        """Statistics of this guild
+
+        Only available on Discord Server"""
+        if ctx.guild is None:
+            return await self.bot.send_cmd_help(ctx)
+        await ctx.trigger_typing()
+        cursor = self.db.commands.find({"guild": ctx.guild.id})
+        data = discord.Embed(
+            description="Command statistics of {0}".format(ctx.guild))
+        data = await self.generate_embed(ctx, data, cursor)
+        try:
+            await ctx.send(embed=data)
+        except discord.Forbidden:
+            await ctx.send("Need permission to embed links")
+
+    @statistics.command(name="total")
+    @commands.is_owner()
+    async def statistics_total(self, ctx):
+        """Total stats of the bot's commands
+
+        Only available to server owner"""
+        await ctx.trigger_typing()
+        cursor = self.db.commands.find()
+        data = discord.Embed(
+            description="Total command statistics")
+        data = await self.generate_embed(ctx, data, cursor)
+        try:
+            await ctx.send(embed=data)
+        except discord.Forbidden:
+            await ctx.send("Need permission to embed links")
+
+
+    async def get_commands(self, cursor, search):
+        """Returns ordered dict of commands from cursor and search string in DB"""
+        commands = {}
+        async for stat in cursor:
+            if stat[search] in commands:
+                commands[stat[search]] += 1
+            else:
+                commands[stat[search]] = 1
+        ordered_commands = collections.OrderedDict(
+            sorted(commands.items(), key=lambda x: x[1], reverse=True))
+        return ordered_commands
+
+    def calc_percentage(self, ordered_commands, total):
+        """Generates ordered dict of percentages of used commands from ordered_commands"""
+        percentages = {}
+        for k, v in ordered_commands.items():
+            percentages[k] = round(100 / total * v)
+        ordered_percentages = collections.OrderedDict(
+            sorted(percentages.items(), key=lambda x: x[1], reverse=True))
+        return ordered_percentages
+
+    async def generate_embed(self, ctx, data, cursor, rank=True):
```

Comment:
Method name is a big vague, rank should probably be a keyword only argument

Reply:
changed to     async def generate_embed(self, ctx, data, cursor, *, rank=True):